crates/biome_js_analyze/src/lint/nursery/no_constant_math_min_max.rs

```diff
@@ -0,0 +1,169 @@
+    /// ```
+    /// ### Valid
+    ///
+    /// ```js
+    /// Math.min(100, Math.max(0, x));
+    /// ```
+    ///
+    pub NoConstantMathMinMax {
+        version: "next",
+        name: "noConstantMathMinMax",
+        sources: &[RuleSource::Clippy("min_max")],
+        recommended: false,
+    }
+}
+
+impl Rule for NoConstantMathMinMax {
+    type Query = Semantic<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+
+        let outer_call = get_math_min_or_max_call(node, model)?;
+
+        let inner_call = get_math_min_or_max_call(
+            outer_call
+                .other_expression_argument
+                .as_js_call_expression()?,
+            model,
+        )?;
+
+        if outer_call.kind == inner_call.kind {
+            return None;
+        }
+
+        match (
+            outer_call.kind,
+            outer_call
+                .constant_argument
+                .partial_cmp(&inner_call.constant_argument),
+        ) {
+            (MinMaxKind::Min, Some(Ordering::Less))
+            | (MinMaxKind::Max, Some(Ordering::Greater)) => Some(()),
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "This "<Emphasis>"Math.min/Math.max"</Emphasis>" combination leads to a constant result."
+            },
+        ))
+    }
+}
+
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+enum MinMaxKind {
+    Min,
+    Max,
+}
+
+impl FromStr for MinMaxKind {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "min" => Ok(MinMaxKind::Min),
+            "max" => Ok(MinMaxKind::Max),
+            _ => Err("Value not supported for math min max kind".to_string()),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct MathMinOrMaxCall {
+    kind: MinMaxKind,
+    constant_argument: f64,
+    other_expression_argument: AnyJsExpression,
+}
+
+fn get_math_min_or_max_call(
+    call_expression: &JsCallExpression,
+    model: &SemanticModel,
+) -> Option<MathMinOrMaxCall> {
+    let callee = call_expression.callee().ok()?.omit_parentheses();
+    let member_expr = AnyJsMemberExpression::cast_ref(callee.syntax())?;
+
+    let member_name = member_expr.member_name()?;
+    let member_name = member_name.text();
+
+    let min_or_max = MinMaxKind::from_str(member_name).ok()?;
+
+    let object = member_expr.object().ok()?.omit_parentheses();
+    let (reference, name) = global_identifier(&object)?;
+
+    if name.text() != "Math" {
+        return None;
+    }
+
+    if model.binding(&reference).is_some() {
+        return None;
+    }
```

Comment:
I suggest merging these two conditions because they are closely related:
```suggestion
    if name.text() != "Math" || model.binding(&reference).is_some() {
        return None;
    }
```